### PR TITLE
Better monorepo support. Use @embroider/addon-blueprint

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
tl;dr: this becomes a serious project 🙃 

Lots of quality-of-life improvements in here, but some tradeoffs that will be less trade-offy as upstream things get resolved.

Support for npm, yarn, and pnpm workspaces, given: https://github.com/embroider-build/addon-blueprint/pull/36

JS Support
 - js support is now _much better_, because I'm using the v2 addon blueprint (with all the flags that have been added for configuring package paths), and modifying the output of that (whereas previously I was building a v2 addon manually) 

TS Support
 - [x] ts support ~~is now _worse_ in the migrator, because I'm waiting on `--typescript` in `@embroider/addon-blueprint` and the ember-cli app blueprint~~
 - [x] will be powered by: https://github.com/embroider-build/addon-blueprint/pull/42 
 
 
 TODO:
 - Support Phased migration
   - [ ] Phase 1: move the tests to their own app
   - [ ] Phase 2: convert the addon to v2
 - [ ] Support downgrading dependencies from the v2 blueprint (rollup-plugin-ts @v2) -- maybe

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Gonna merge since this PR has gotten ridiculous, C.I., and the aforementioned features can be developed in PRs that are actually easy to review.
